### PR TITLE
feat: finish compression-related functionality in the import process (on top of #28)

### PR DESCRIPTION
**Pull request by doudou**
Essentially adding import of compressed Roby logs, and compression of the logs themselves. The generated roby
index is not compressed (usually very small)

### CODE REVIEW

File: lib/syskit/log/datastore/import.rb [hunk 1]
```diff
@@ -324,26 +334,29 @@ def copy_roby_event_log(event_log_path, roby_sql_index)
...
                 true
             rescue Roby::DRoby::Logfile::TruncatedFileError => e
                 @reporter.warn e.message
                 @reporter.warn "truncating Roby log file"
-                index_io.rewind
-                Roby::DRoby::Logfile::Index.write_header(
-                    index_io, pos, in_stat.mtime
-                )
+                if index_io
+                    index_io&.rewind
```
Comment (jhonasiv) [inline lib/syskit/log/datastore/import.rb]: no need for the `&` here, index_io is already valid

File: lib/syskit/log/zstd_io.rb [hunk 2]
```diff
@@ -12,16 +12,31 @@ class ZstdIO
...
             end
 
             def path
                 @io.path
             end
 
+            def rewind
+                @io.rewind
+                @tell = 0
+                @buffer = +""
```
Comment (jhonasiv) [inline lib/syskit/log/zstd_io.rb:27]: what's the point of this +?

File: lib/syskit/log/datastore/import.rb (hunk 1 above)
Comment (doudou) [reply lib/syskit/log/datastore/import.rb]: fixed

File: lib/syskit/log/zstd_io.rb (hunk 2 above)
Comment (doudou) [reply lib/syskit/log/zstd_io.rb:27]: String literals are frozen (read-only), the + makes it read-write

**Review by jhonasiv — APPROVED** [review]

**Pull request merged**